### PR TITLE
User Registration

### PR DESCRIPTION
**Pull request by sahedw**
- create dto for user registration
- create methods in controller and new service
- write integration tests
- create signup page and axios request for registration
- write the function in a way that the user gets redirected directly on the page logged in if registration successful
- write custom exception for already existing username and service exception test
- create frontend form with request and hide password option

### CODE REVIEW

File: backend/src/test/java/com/github/sahedw/backend/security/FoodSpotUserServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        foodSpotUserServiceReal.signUp(newUserSignUp);
+        // WHEN
+        when(foodSpotUserRepoMocked.findByUsername("franz")).thenReturn(Optional.of(newUser));
+        FoodSpotUserForSignUp secondUser = new FoodSpotUserForSignUp("franz", "franz2");
+
+        // Assert
+        assertThrows(UsernameAlreadyExistsException.class, () -> foodSpotUserServiceReal.signUp(secondUser));
+    }
+
+}
```
Comment (Ben-21) [inline backend/src/test/java/com/github/sahedw/backend/security/FoodSpotUserServiceTest.java:68]: You already know the problem, right? 😂

File: frontend/src/icons/SeePassword.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+                </svg>
+            :
+                <svg onClick={onShowPassword} width={size} height={size} viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg" className={className}>
+                    <path fill="#000000" d="M11.83 9L15 12.16V12a3 3 0 0 0-3-3h-.17m-4.3.8l1.55 1.55c-.05.21-.08.42-.08.65a3 3 0 0 0 3 3c.22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53a5 5 0 0 1-5-5c0-.79.2-1.53.53-2.2M2 4.27l2.28 2.28l.45.45C3.08 8.3 1.78 10 1 12c1.73 4.39 6 7.5 11 7.5c1.55 0 3.03-.3 4.38-.84l.43.42L19.73 22L21 20.73L3.27 3M12 7a5 5 0 0 1 5 5c0 .64-.13 1.26-.36 1.82l2.93 2.93c1.5-1.25 2.7-2.89 3.43-4.75c-1.73-4.39-6-7.5-11-7.5c-1.4 0-2.74.25-4 .7l2.17 2.15C10.74 7.13 11.35 7 12 7Z"></path>
+                </svg>
+            }
+    </>
+
+    )
+}
```
Comment (Ben-21) [inline frontend/src/icons/SeePassword.tsx:21]: ...got you 🙃

**Review by Ben-21 — APPROVED** [review]
Really, really nice work!
And you are extremely fast (at least for my understanding) 💪 🚀 👌

File: backend/src/test/java/com/github/sahedw/backend/security/FoodSpotUserServiceTest.java (hunk 1 above)
Comment (sahedw) [reply backend/src/test/java/com/github/sahedw/backend/security/FoodSpotUserServiceTest.java:68]: Whoops, yeah I can guess 😂

**Pull request merged**